more configurable release options

peter-mouland opened the pull request:
allow use to specify whether the release command should release to git, gh-pages and/or s3 using an array in the config  `release: ['git','gh-pages','s3']`

changes is backwards compatible.  if a string is found, it assumes you also want to release to git and gh-pages (i.e. same as pre v1.2).

fixes #78

peter-mouland commented:
fixes #95

behaghel commented on tasks/release.js:
```diff
@@ -1,15 +1,25 @@
+
 function ghPagesRelease(message){
+    getConfig();
+    if (component.release.indexOf('s3') <0){
```
s3? gh-pages rather, isn't it?

behaghel commented on tasks/release.js:
```diff
@@ -20,14 +30,14 @@ function ghPagesRelease(message){
 }
 
 function s3(version){
```
All those functions could actually take the config as a parameter. I don't think we should do it as part of this PR but I see an opportunity to define an interface function(message, config){} to make this easily extensible to other platforms.

peter-mouland replied:
yeah, i've been thinking about this for a while too.

behaghel commented:
For me it's a **ship it**.

peter-mouland commented:
cheers for the review - i've added tests and fix the bug you spotted :)

behaghel commented:
:+1: well done on the testing!

peter-mouland commented:
@behaghel just to let you know - I've added another change to this PR.
I thought, if `release` can be an array, why am i being so verbose with `build` options.
So i've changed this to an array also, which makes the config file smaller.
the change also has some code so not to break old config styles.

I'll try be more careful with breaking changes in the future i think and proper semanic-versioning

behaghel commented:
Nice move! Again it seems to shout for an extensible approach where say `browserify` is a reference to a script that doesn't have to be part of this repo (could be local to the client project, e.g. `build/browserify.js`) that hooks itself to a particular step in the build life cycle.

peter-mouland commented:
a couple of good spots, fixed and added tests :)
Interesting thought about making this repo more light-weight and extensible. I think that's worth opening an issue to discuss in more detail.

behaghel commented:
I think it's ready to be merged. I'll create an issue re extensible tasks.
